apps/trading/pages/markets/[marketId].page.tsx

```diff
@@ -0,0 +1,89 @@
+import { gql, useQuery } from '@apollo/client';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import { ReactNode } from 'react';
+import { Market, MarketVariables } from './__generated__/Market';
+
+// Top level page query
+const MARKET_QUERY = gql`
+  query Market($marketId: ID!) {
+    market(id: $marketId) {
+      id
+      name
+      trades {
+        id
+        price
+        size
+        createdAt
+      }
+    }
+  }
+`;
+
+const MarketPage = () => {
+  const { query } = useRouter();
+  const { data, loading, error } = useQuery<Market, MarketVariables>(
+    MARKET_QUERY,
+    {
+      variables: { marketId: query.marketId as string },
+    }
+  );
+
+  if (loading || !data) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <>
+      {error ? (
+        <div>Something went wrong: {error.message}</div>
+      ) : (
```

Comment:
(followup) non-blocking: would be nice to wrap this is a query component so we don't implement this a million times

Reply:
Done! I've added a PageQueryContainer component which takes the query and query options as props, handles consistent loading and error states and passes the resulting data as a render prop.

I've also added it to the market list page.